cmd/registry/patch/patch_test.go

```diff
@@ -1009,64 +1011,82 @@ func TestMessageArtifactPatches(t *testing.T) {
 			},
 		},
 	}
-	ctx := context.Background()
-	registryClient, _ := grpctest.SetupRegistry(ctx, t, "patch-message-artifact-test", []seeder.RegistryResource{
-		&rpc.ApiSpec{
-			Name: root + "/apis/a/versions/v/specs/s",
-		},
-	})
 
-	for _, test := range tests {
-		t.Run(test.artifactID, func(t *testing.T) {
-			b, err := os.ReadFile(test.yamlFile)
-			if err != nil {
-				t.Fatalf("%s", err)
-			}
-			err = applyArtifactPatchBytes(ctx, registryClient, b, root, "patch.yaml")
-			if err != nil {
-				t.Fatalf("%s", err)
-			}
-			var collection string
-			if test.parent != "" {
-				collection = root + "/" + test.parent + "/artifacts/"
-			} else {
-				collection = root + "/artifacts/"
-			}
-			artifactName, err := names.ParseArtifact(collection + test.artifactID)
-			if err != nil {
-				t.Fatalf("%s", err)
-			}
-			err = visitor.GetArtifact(ctx, registryClient, artifactName, true,
-				func(ctx context.Context, artifact *rpc.Artifact) error {
-					contents, err := getArtifactMessageContents(artifact)
+	storageTypes := []struct {
+		mimeBase string
+		ctx      context.Context
+	}{
+		{"application/octet-stream", context.Background()},
+		{"application/yaml", SetStoreArchivesAsYaml(context.Background())},
```

Comment:
This is our main test that this works, right? Verifying that all artifact types can be stored and read in both formats? (just thinking out loud... LGTM)

Reply:
> > after forcing the archive storage to default to yaml
> 
> Just wondering... how did you do that? By temporarily hard-coding [this](https://github.com/apigee/registry/blob/8dab302177542a97f479154ae1079c8385cb8b95/cmd/registry/cmd/apply/apply.go#L32) to true?

Close, but [here](https://github.com/apigee/registry/blob/8dab302177542a97f479154ae1079c8385cb8b95/cmd/registry/patch/artifact.go#L113)

Reply:
Yes